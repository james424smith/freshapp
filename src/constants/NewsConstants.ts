export const READ_MORE_NEWS = "ReadMoreNews";
/**
 * In order to get the types work we need to declare the type of the constant as its value as well.
 * (This is a limitation of flow that doesn't accept a constant's value as the type of a variable)
 * So both strings entered in a constant declaration need to be in sync.
 */
export const NEWSLETTERS: "Newsletters" = "Newsletters";
export const PRESS_RELEASES: "PressReleases" = "PressReleases";
