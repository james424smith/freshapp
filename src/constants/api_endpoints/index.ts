export const assignmentsEndpoint = (
  baseUrl: string,
  seafarerId: number
): string => `${baseUrl}/seafarers/${seafarerId}/all-assignments`;

export const imprintEndpoint = (baseUrl: string): string =>
  `${baseUrl}/imprint`;

export const seafarerEndpoint = (baseUrl: string, seafarerId: number): string =>
  `${baseUrl}/seafarers/${seafarerId}`;

export const newsEndpoint = (baseUrl: string): string => `${baseUrl}/news`;

export const seafarerDocumentsEndpoint = (
  baseUrl: string,
  seafarerId: number
): string => `${baseUrl}/seafarers/${seafarerId}/seafarersdocuments`;

export const seafarerDocumentsFileEndpoint = (
  baseUrl: string,
  seafarerId: number,
  documentId?: string | number,
  documentCounter?: number
): string => {
  if (documentId && documentCounter) {
    return `${baseUrl}/seafarers/${seafarerId}/seafarersdocuments/${documentId}/${documentCounter}`;
  }
  throw Error("documentId and documentCounter cann be undefined");
};

export const signUpEndpoint = (baseUrl: string): string =>
  `${baseUrl}/register`;

export const termsAndConditionsEndpoint = (baseUrl: string): string =>
  `${baseUrl}/termsconditions`;

export const privacyPolicyEndpoint = (baseUrl: string): string =>
  `${baseUrl}/privacypolicy`;

export const workingClothesEndpoint = (
  baseUrl: string,
  seafarerId: number
): string => `${baseUrl}/seafarers/${seafarerId}/workingclothes`;

export const seaServiceEndpoint = (
  baseUrl: string,
  seafarerId: number
): string => `${baseUrl}/seafarers/${seafarerId}/seaservicerecords`;

export const payslipEndpoint = (baseUrl: string, seafarerId: number): string =>
  `${baseUrl}/seafarers/${seafarerId}/salaries`;

export const checkVersionEndpoint = (baseUrl: string): string =>
  `${baseUrl}/versions`;

export const payslipDocumentsFileEndpoint = (
  baseUrl: string,
  seafarerId: number,
  vesselCode?: string,
  payslipId?: string | number
): string => {
  if (vesselCode && payslipId) {
    return `${baseUrl}/seafarers/${seafarerId}/salaries/${vesselCode}/${payslipId}`;
  }
  throw Error("vesselCode and payslipId cannot be undefined");
};

export const allotmentsEndpoint = (
  baseUrl: string,
  seafarerId: number
): string => `${baseUrl}/seafarers/${seafarerId}/allotments`;

export const notificationsEndpoint = (
  baseUrl: string,
  seafarerId: number
): string => `${baseUrl}/seafarers/${seafarerId}/notifications`;

export const notificationReadEndpoint = (baseUrl: string): string =>
  `${baseUrl}/notificationread`;

export const seafarerBadgesEndpoint = (
  baseUrl: string,
  seafarerId: number
): string => `${baseUrl}/seafarers/${seafarerId}/badge`;

export const contactEndpoint = (baseUrl: string, seafarerId: number): string =>
  `${baseUrl}/seafarers/${seafarerId}/contacts`;

export const flightsEndpoint = (baseUrl: string, seafarerId: number): string =>
  `${baseUrl}/seafarers/${seafarerId}/flights`;

export const covidDocumentEndpoint = (baseUrl: string): string =>
  `${baseUrl}/covidfile/covid`;

export const validateWebsocketEndpoint = (
  baseUrl: string,
  seafarerId: number
): string => `${baseUrl}/chat/seafarers/${seafarerId}/validate`;

export type endpointOptions =
  | "assignment"
  | "imprint"
  | "seafarer"
  | "signup"
  | "crewManager"
  | "manningAgent"
  | "termsAndConditions"
  | "privacyPolicy"
  | "news"
  | "workingClothes"
  | "seafarerDocuments"
  | "seaService"
  | "payslip"
  | "allotments"
  | "notifications"
  | "notificationRead"
  | "seafarerBadges"
  | "seafarerDocumentsFileEndpoint"
  | "contact"
  | "flights"
  | "payslipDocumentsFile"
  | "seafarerDocumentsFile"
  | "covid"
  | "validateWebsocket"
  | "versionCheck";
