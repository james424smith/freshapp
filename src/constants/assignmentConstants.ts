export const CURRENT: "current" = "current";
export const NEXT: "next" = "next";
