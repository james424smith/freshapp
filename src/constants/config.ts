export declare const __DEV__: boolean;
import Config from "react-native-ultimate-config";

/* So babel needs to know that there is a change to the file where the react-native-dotenv is being imported.
 * That's why import the env variables here and export them to whenever we use them for. As soon as change the .env file,
 * we need to come here and make any small -not-breaking- change (such as adding a space to the console.log string) in order to get the updated values
 */

export default () => {
  return {
    ENV: Config.ENV || "DUMMY",
    API_BASE_URL: Config.API_BASE_URL,
    IMAGES_PREFIX_URL: Config.IMAGES_PREFIX_URL,
    WEBSOCKET_BASE_URL: Config.WEBSOCKET_BASE_URL,
    GOOGLE_MAPS_API_KEY: Config.GOOGLE_MAPS_API_KEY,
    SSL_EXPIRATION_WARNING: Config.SSL_EXPIRATION_WARNING,
  };
};
