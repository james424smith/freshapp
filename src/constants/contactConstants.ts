export const CREW_MANAGER: "Crew Manager" = "Crew Manager";
export const PORT_AGENT: "Port Agent" = "Port Agent";
export const MANNING_AGENT: "Manning Agent" = "Manning Agent";
