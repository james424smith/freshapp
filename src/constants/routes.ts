export const MAIN_APP_SCREEN_ROUTE = "Main";

// Main screens
export const ASSIGNMENT_ROUTE = "AssignmentStack";
export const FLIGHT_ROUTE = "FlightsStack";
export const CHAT_ROUTE = "MainChat";
export const CHAT_HISTORY_ROUTE = "ChatHistory";
export const CHAT_ROOM_ROUTE = "ChatRoom";
export const ASSIGNMENT_PROFILE_ROUTE = "AssignmentSeafarerProfile";
export const LATEST_RELEASES_ROUTE = "LatestReleases";

// Single pages
export const AUTH_LOADING_STACK_ROUTE = "AuthLoading";
export const TERMS_AND_CONDITIONS_ROUTE = "TermsAndConditionsAuth";
export const PRIVACY_POLICY_ROUTE = "PrivacyPolicyAuth";
export const PASSCODE_ROUTE = "PassCode";
export const NEWS_ROUTE = "News";
export const EMPLOYMENT_OFFER_ROUTE = "EmploymentOffer";
export const PROFILE_ROUTE = "SeafarerDetails";
export const WORKING_CLOTHES_ROUTE = "WorkingClothes";
export const VESSEL_TRACKER_ROUTE = "VesselTracker";
export const SEA_SERVICE_RECORDS_ROUTE = "SeaServiceRecords";
export const DOCUMENTS_ROUTE = "SeafarerDocuments";
export const READ_MORE_NEWS_ROUTE = "ReadMoreNews";
export const SALARY_ROUTE = "Salary";
export const READ_MORE_PAYSLIP = "ReadMorePayslip";
export const CONTACT_ROUTE = "Contact";
export const IMPRINT_ROUTE = "Imprint";
export const CREW_PORTAL_ROUTE = "Crew Portal";
export const SUPPORT_ROUTE = "AppSupport";
export const COVID_DOCUMENT_ROUTE = "CovidDocument";
export const SETTINGS_ROUTE = "Settings";

//Auth
export const FORCE_UPDATE_PASSWORD_ROUTE = "ForceUpdatePassword";
export const SIGN_IN_ROUTE = "SignIn";
export const SIGN_UP_ROUTE = "SignUp";
export const FORGOT_PASSWORD_ROUTE = "ForgotPassword";
export const RESET_PASSWORD_ROUTE = "ResetPassword";
export const ACCOUNT_VERIFICATION_ROUTE = "AccountVerification";
export const FAQ_ROUTE = "FAQ";
export const SWIPE_SCREEN_ROUTE = "SwipeNavigator";
