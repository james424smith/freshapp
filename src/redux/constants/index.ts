export * from "./seafarerDetailsConstants";
export * from "./assignmentDetailsConstants";
export * from "./imprintDetailsConstants";
export * from "./newsConstants";
export * from "./termsAndConditionsConstants";
export * from "./workingClothesConstants";
export * from "./privacyPolicyConstants";
export * from "./seafarerDocumentsConstants";
export * from "./seaServiceDetailsConstants";
export * from "./notificationsConstants";
export * from "./salaryDetailsConstants";
export * from "./contactDetailsConstants";
export * from "./flightsDetailsConstants";
export * from "./websocketConstants";
export * from "./covidDocumentConstants";
export * from "./darkModeOptionsConstants";
export const CLEAR_ALL_STATE = "CLEAR_ALL_STATE";
