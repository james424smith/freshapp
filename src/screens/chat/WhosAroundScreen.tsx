import React, { useState, useEffect } from "react";
import {
  View,
  Alert,
  TouchableOpacity,
  Platform,
  PermissionsAndroid,
  ScrollView,
} from "react-native";
import Text from "../../components/StyledText";
import Colors from "../../constants/styles/colors";
import ToggleSwitch from "toggle-switch-react-native";
import { CHAT_HISTORY_ROUTE } from "../../constants/routes";
import Geolocation from "react-native-geolocation-service";
import styles from "../../styles/ChatStyles";
import { useSelector, useDispatch } from "react-redux";
import { setWebsocketData } from "../../redux/actions";
import { Auth } from "aws-amplify";
import {
  UpdateWhoIsAroundRequest,
  UPDATE_WHO_IS_AROUND,
} from "../../interfaces";
import { timestampToLocalString } from "../../common/dateUtils";
import { Picker } from "@react-native-picker/picker";
import { IRootReducerType } from "../../redux/reducers";
import CloseSvg from "../../../assets/icons/bottom_navigation/close.svg";
import { useNavigation, useTheme } from "@react-navigation/native";
import LoadingScreen from "../../components/LoadingScreen";

const useShouldRedirect = () => {
  const [shouldRedirect, setShouldRedirect] = useState<boolean>(false);
  const isOnline = useSelector<IRootReducerType, boolean>(
    ({ websocketReducer }) => websocketReducer.isOnline
  );
  const navigation = useNavigation();
  useEffect(() => {
    if (isOnline === true && shouldRedirect) {
      setShouldRedirect(false);
      navigation.navigate(CHAT_HISTORY_ROUTE);
    }
  }, [isOnline, shouldRedirect, navigation]);

  return { setShouldRedirect, isOnline };
};

const CloseIcon = () => {
  const navigation = useNavigation();
  const theme = useTheme();
  return (
    <TouchableOpacity
      testID={"close-icon"}
      onPress={() => navigation.navigate(CHAT_HISTORY_ROUTE)}
      style={styles.closeIconContainer}
    >
      <CloseSvg
        fill={theme.colors.blackAndWhite}
        style={styles.closeIconStyle}
      />
    </TouchableOpacity>
  );
};

const TextRenderer = ({
  selectedMode,
  onlineUntil,
}: {
  selectedMode: boolean;
  onlineUntil: number;
}) => {
  const theme = useTheme();
  return (
    <View style={styles.whosAroundContainer}>
      {!selectedMode ? (
        <View>
          <Text
            style={{
              ...styles.whosAroundStatusText,
              color: theme.colors.text,
            }}
          >
            Who's around is disabled
          </Text>
          <Text
            style={{
              ...styles.whosAroundParagraph,
              color: theme.colors.text,
            }}
          >
            Turn on to check who's around and chat
          </Text>
          <Text
            style={{
              ...styles.whosAroundParagraph,
              color: theme.colors.text,
            }}
          >
            with your fellow seafarers
          </Text>
        </View>
      ) : (
        <View>
          <Text
            style={{
              ...styles.whosAroundStatusText,
              color: theme.colors.text,
            }}
          >
            You are now visible to fellow seafarers around you.
          </Text>
          <Text
            style={{
              ...styles.whosAroundParagraph,
              color: theme.colors.text,
            }}
          >
            You will be active until{" "}
          </Text>
          <Text
            style={{
              ...styles.whosAroundStatusText,
              color: theme.colors.text,
            }}
          >
            {timestampToLocalString(onlineUntil)}
          </Text>
        </View>
      )}
    </View>
  );
};

const WhosAroundScreen = () => {
  const theme = useTheme();
  const logoSize = 100;

  const onlineUntil = useSelector<IRootReducerType, number>(
    ({ websocketReducer }) => websocketReducer.onlineUntil
  );
  const dispatch = useDispatch();
  const [chosenHours, setChosenHours] = useState<string>("0h");
  const [chosenMinutes, setChosenMinutes] = useState<string>("40m");
  const [chosenIndexHours, setChosenIndexHours] = useState<number>(0);
  const [chosenIndexMinutes, setChosenIndexMinutes] = useState<number>(0);
  const [enablingChat, setEnablingChat] = useState<boolean>(false);
  const { isOnline, setShouldRedirect } = useShouldRedirect();
  const geoSuccess = async (position: any) => {
    const user = await Auth.currentUserInfo();
    const minutesOnline = chosenIndexHours * 60 + chosenIndexMinutes;

    const whoIsAroundData =
      isOnline === true
        ? {
            seafarerId: `${user.username}`,
            pin: {
              lat: position.coords.latitude,
              lon: position.coords.longitude,
            },
          }
        : {
            seafarerId: `${user.username}`,
            pin: {
              lat: position.coords.latitude,
              lon: position.coords.longitude,
            },
            onlineUntil: minutesOnline,
          };
    dispatch(
      setWebsocketData({
        messageType: UPDATE_WHO_IS_AROUND,
        payload: whoIsAroundData,
      })
    );
    setEnablingChat(false);
  };

  const geoFailure = async (err: any) => {
    console.error("Error while getting geolocation", err);
    const user = await Auth.currentUserInfo();
    const whoIsAroundData: UpdateWhoIsAroundRequest = {
      seafarerId: `${user.username}`,
      pin: {
        lat: 0,
        lon: 0,
      },
      onlineUntil: 0,
    };
    dispatch(
      setWebsocketData({
        messageType: UPDATE_WHO_IS_AROUND,
        payload: whoIsAroundData,
      })
    );
    Alert.alert(
      "We couldn't get your location at the moment. Please check your settings in order to be online"
    );
    setEnablingChat(false);
  };

  const handleGetGeolocation = async () => {
    setEnablingChat(true);
    if (Platform.OS === "android") {
      await PermissionsAndroid.request(
        PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION
      );
      await PermissionsAndroid.request(
        PermissionsAndroid.PERMISSIONS.ACCESS_COARSE_LOCATION
      );
    } else {
      await Geolocation.requestAuthorization("always");
    }
    Geolocation.getCurrentPosition(geoSuccess, geoFailure, {
      enableHighAccuracy: true,
      timeout: 10000,
    });
  };

  const handleToggleSelection = async () => {
    setShouldRedirect(true);
    if (isOnline) {
      const user = await Auth.currentUserInfo();
      const whoIsAroundData: UpdateWhoIsAroundRequest = {
        seafarerId: `${user.username}`,
        pin: {
          lat: 0,
          lon: 0,
        },
        onlineUntil: -1,
      };
      dispatch(
        setWebsocketData({
          messageType: UPDATE_WHO_IS_AROUND,
          payload: whoIsAroundData,
        })
      );
    } else {
      await handleGetGeolocation();
    }
  };

  const renderTimeDropdown = () => {
    const dropDownStyles =
      Platform.OS === "android"
        ? {
            color: "#000",
            style: {
              color: theme.colors.blackAndWhite,
            },
          }
        : { color: theme.colors.text };
    if (!isOnline) {
      return (
        <View style={styles.timeSelectionRow}>
          <View style={styles.spaceView} />
          <View style={styles.pickerView}>
            <Picker
              mode="dropdown"
              selectedValue={chosenHours}
              style={dropDownStyles.style}
              onValueChange={(itemValue) => {
                setChosenHours(`${itemValue}`);
                setChosenIndexHours(Number(itemValue));
              }}
            >
              {["0", "1", "2", "3", "4", "5"].map((hours) => (
                <Picker.Item
                  label={`${hours}h`}
                  color={dropDownStyles.color}
                  value={hours}
                  key={hours}
                />
              ))}
            </Picker>
          </View>

          <View style={styles.pickerView}>
            <Picker
              mode="dropdown"
              selectedValue={chosenMinutes}
              style={dropDownStyles.style}
              onValueChange={(itemValue) => {
                setChosenMinutes(`${itemValue}`);
                setChosenIndexMinutes(Number(itemValue));
              }}
            >
              {["00", "30"].map((minutes) => (
                <Picker.Item
                  key={minutes}
                  color={dropDownStyles.color}
                  label={`${minutes}m`}
                  value={minutes}
                />
              ))}
            </Picker>
          </View>

          <View style={styles.spaceView} />
        </View>
      );
    }
    return null;
  };

  return (
    <ScrollView
      style={styles.scrollView}
      contentContainerStyle={styles.scrollViewContainer}
      scrollEnabled={false}
    >
      {enablingChat ? (
        <View style={styles.root}>
          <LoadingScreen logoWidthSize={logoSize} logoHeightSize={logoSize} />
        </View>
      ) : (
        <>
          <CloseIcon />
          <View style={styles.isOnlineView}>
            <View>
              {
                <TextRenderer
                  onlineUntil={onlineUntil}
                  selectedMode={isOnline}
                />
              }
            </View>
            <View style={styles.toggleContainer}>
              <ToggleSwitch
                data-testid={"toggle"}
                isOn={isOnline}
                onColor={Colors.marlowBlue}
                offColor={Colors.grey}
                size="large"
                onToggle={handleToggleSelection}
              />
            </View>
            <View>{renderTimeDropdown()}</View>
          </View>
        </>
      )}
    </ScrollView>
  );
};

export default WhosAroundScreen;
