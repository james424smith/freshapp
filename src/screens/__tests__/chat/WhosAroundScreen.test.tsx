import React from "react";
import WhosAroundScreen from "../../chat/WhosAroundScreen";
import {
  render,
  cleanup,
  waitFor,
  fireEvent,
} from "@testing-library/react-native";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { Settings } from "luxon";
import MockDate from "mockdate";
import Amplify, { Analytics, Auth } from "aws-amplify";
import { Platform } from "react-native";
import { useNavigation } from "@react-navigation/native";
import { CHAT_HISTORY_ROUTE } from "../../../constants/routes";

Amplify.configure({
  aws_project_region: "eu-west-1",
  aws_cognito_identity_pool_id:
    "eu-west-1:0700dd4d-d3f1-47e8-aeca-c03a916aba97",
  aws_cognito_region: "eu-west-1",
  aws_user_pools_id: "eu-west-1_VdCn6sWLK",
  aws_user_pools_web_client_id: "3uc4kul2beqktthivc578jutfm",
  oauth: {},
  aws_mobile_analytics_app_id: "33c953d8b8a04972904e65148ea64665",
  aws_mobile_analytics_app_region: "eu-west-1",
});

describe("WhosAroundScreen snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui when isOnline is true and platform is ios", async () => {
    Auth.currentAuthenticatedUser = jest.fn().mockResolvedValue("12345");
    Analytics.updateEndpoint = jest.fn().mockResolvedValue("12345");
    MockDate.set(1574237559000);

    Platform.OS = "ios";
    Settings.now = () => new Date().valueOf();

    const mockedState = {
      websocketReducer: {
        isOnline: true,
        onlineUntil: 1574237659000,
      },
    };
    const mockedStore = configureMockStore();

    const { toJSON, getByTestId } = render(
      <Provider store={mockedStore(mockedState)}>
        <WhosAroundScreen />
      </Provider>
    );

    fireEvent.press(getByTestId("close-icon"));
    expect(useNavigation().navigate).toHaveBeenCalledWith(CHAT_HISTORY_ROUTE);

    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
    MockDate.reset();
  });
  it("should render correct ui when isOnline is true and platform is adnroid", async () => {
    MockDate.set(1574237559000);

    Platform.OS = "android";
    Settings.now = () => new Date().valueOf();

    const mockedState = {
      websocketReducer: {
        isOnline: true,
        onlineUntil: 1574237659000,
      },
    };
    const mockedStore = configureMockStore();

    const { toJSON } = render(
      <Provider store={mockedStore(mockedState)}>
        <WhosAroundScreen />
      </Provider>
    );

    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
    MockDate.reset();
  });
});
