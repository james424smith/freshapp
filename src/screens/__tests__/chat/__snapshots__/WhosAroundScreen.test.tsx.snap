// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhosAroundScreen snapshot test should render correct ui when isOnline is true and platform is adnroid 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  scrollEnabled={false}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "flex-end",
          "marginTop": 75,
          "opacity": 1,
        }
      }
      testID="close-icon"
    >
      <<?xml version="1.0" encoding="iso-8859-1"?>
<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 512 512" style="enable-background:new 0 0 512 512;" xml:space="preserve">
<g>
	<g>
		<path d="M505.943,6.058c-8.077-8.077-21.172-8.077-29.249,0L6.058,476.693c-8.077,8.077-8.077,21.172,0,29.249
			C10.096,509.982,15.39,512,20.683,512c5.293,0,10.586-2.019,14.625-6.059L505.943,35.306
			C514.019,27.23,514.019,14.135,505.943,6.058z"/>
	</g>
</g>
<g>
	<g>
		<path d="M505.942,476.694L35.306,6.059c-8.076-8.077-21.172-8.077-29.248,0c-8.077,8.076-8.077,21.171,0,29.248l470.636,470.636
			c4.038,4.039,9.332,6.058,14.625,6.058c5.293,0,10.587-2.019,14.624-6.057C514.018,497.866,514.018,484.771,505.942,476.694z"/>
	</g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
</svg>

        style={
          Object {
            "height": 53.5,
            "marginRight": 37.5,
            "width": 34,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "flexDirection": "column",
          "flexGrow": 1,
          "justifyContent": "space-around",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignSelf": "center",
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 40,
                    "fontWeight": "900",
                    "paddingBottom": 30,
                    "textAlign": "center",
                  },
                ]
              }
            >
              You are now visible to fellow seafarers around you.
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                    "textAlign": "center",
                  },
                ]
              }
            >
              You will be active until
               
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 40,
                    "fontWeight": "900",
                    "paddingBottom": 30,
                    "textAlign": "center",
                  },
                ]
              }
            >
              20.11.19    08:14
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#005AA5",
                "borderRadius": 20,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 20,
                "width": 70,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderRadius": 15,
                  "elevation": 1.5,
                  "height": 30,
                  "justifyContent": "center",
                  "left": 0,
                  "margin": 4,
                  "position": "absolute",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.2,
                  "shadowRadius": 2.5,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 30,
                }
              }
            />
          </View>
        </View>
      </View>
      <View />
    </View>
  </View>
</RCTScrollView>
`;

exports[`WhosAroundScreen snapshot test should render correct ui when isOnline is true and platform is ios 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
  scrollEnabled={false}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "flex-end",
          "marginTop": 75,
          "opacity": 1,
        }
      }
      testID="close-icon"
    >
      <<?xml version="1.0" encoding="iso-8859-1"?>
<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 512 512" style="enable-background:new 0 0 512 512;" xml:space="preserve">
<g>
	<g>
		<path d="M505.943,6.058c-8.077-8.077-21.172-8.077-29.249,0L6.058,476.693c-8.077,8.077-8.077,21.172,0,29.249
			C10.096,509.982,15.39,512,20.683,512c5.293,0,10.586-2.019,14.625-6.059L505.943,35.306
			C514.019,27.23,514.019,14.135,505.943,6.058z"/>
	</g>
</g>
<g>
	<g>
		<path d="M505.942,476.694L35.306,6.059c-8.076-8.077-21.172-8.077-29.248,0c-8.077,8.076-8.077,21.171,0,29.248l470.636,470.636
			c4.038,4.039,9.332,6.058,14.625,6.058c5.293,0,10.587-2.019,14.624-6.057C514.018,497.866,514.018,484.771,505.942,476.694z"/>
	</g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
</svg>

        style={
          Object {
            "height": 53.5,
            "marginRight": 37.5,
            "width": 34,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "flexDirection": "column",
          "flexGrow": 1,
          "justifyContent": "space-around",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignSelf": "center",
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 40,
                    "fontWeight": "900",
                    "paddingBottom": 30,
                    "textAlign": "center",
                  },
                ]
              }
            >
              You are now visible to fellow seafarers around you.
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 24,
                    "textAlign": "center",
                  },
                ]
              }
            >
              You will be active until
               
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 40,
                    "fontWeight": "900",
                    "paddingBottom": 30,
                    "textAlign": "center",
                  },
                ]
              }
            >
              20.11.19    08:14
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#005AA5",
                "borderRadius": 20,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 20,
                "width": 70,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderRadius": 15,
                  "elevation": 1.5,
                  "height": 30,
                  "justifyContent": "center",
                  "left": 0,
                  "margin": 4,
                  "position": "absolute",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.2,
                  "shadowRadius": 2.5,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 30,
                }
              }
            />
          </View>
        </View>
      </View>
      <View />
    </View>
  </View>
</RCTScrollView>
`;
